dtfs-central-api/src/services/tfm/deal-cancellation.service.ts

```diff
@@ -64,6 +64,17 @@ export class DealCancellationService {
         auditDetails,
       });
 
```

Comment:
@MarRobSoftwire this function is starting to get rather large -- when we look to update tfm status could we refactor this down into

```
submitDealCancellation(){
    // Can this logic for audit details be abstracted at all?
    const user = await TfmUsersRepo.findOneUserById(auditDetails.id);

    if (!user) {
      throw new InvalidAuditDetailsError(`Supplied auditDetails 'id' ${auditDetails.id.toString()} does not correspond to a valid user`);
    }}
    // end audit detail abstraction comment

    if(DealCancellationService.isDealCancellationInFuture(cancellation)) {
        DealCancellationService.handleFutureDealCancellation() // this would contain all teh logic of updating portal/tfm deal status as well as scheduling the cancellation (which, actually should just be driven by the deal status & cancellation date?) 
    } else {
        DealCancellationService.handleCancelDeal()
    } // The else here might add clarity as to the condition of the code below it running

```

Reply:
This sounds good - had a little play (wont do now to keep PR slim) but think this could work
```
generateCancellationActivityFromAuditDetails
submitDealCancellationInFuture
submitDealCancellationInPast
```

leaving something like
```
  public static async submitDealCancellation(
    dealId: ObjectId | string,
    cancellation: TfmDealCancellation,
    auditDetails: TfmAuditDetails,
  ): Promise<TfmDealCancellationResponse> {
    const effectiveFromDate = toDate(cancellation.effectiveFrom);
    const endOfToday = endOfDay(new Date());

    const dealCancellationIsInFuture = isAfter(effectiveFromDate, endOfToday);

    if (dealCancellationIsInFuture) {
      return await this.submitDealCancellationInFuture(dealId, cancellation, auditDetails);
    }
    return await this.submitDealCancellationInPast(dealId, cancellation, auditDetails);
  }
```

Reply:
Any reason we're not using `isDealCancellationInFuture` for the boolean naming?

Looking a lot nicer! Great job.

I'm wondering about `submitDealCancellationInFuture` and `submitDealCancellationInPast` as names -- I'm not overly keen but maybe it is better than `handleFutureDealCancellation`? I think `submitFutureDealCancellation` does add ambiguity as to what is in the future, so maybe this is a good name!